pkg/reconciler/pubsub/pubsub.go

```diff
@@ -0,0 +1,212 @@
+
+	resourceGroup = "pubsubs.events.cloud.run"
+)
+
+// Reconciler is the controller implementation for the PubSub source.
+type Reconciler struct {
+	*reconciler.PubSubBase
+
+	// pubsubLister for reading pubsubs.
+	pubsubLister listers.PubSubLister
+}
+
+// Check that we implement the controller.Reconciler interface.
+var _ controller.Reconciler = (*Reconciler)(nil)
+
+// Reconcile implements controller.Reconciler
+func (c *Reconciler) Reconcile(ctx context.Context, key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+
+	// Get the PubSub resource with this namespace/name
+	original, err := c.pubsubLister.PubSubs(namespace).Get(name)
+	if apierrs.IsNotFound(err) {
+		// The PubSub resource may no longer exist, in which case we stop processing.
+		runtime.HandleError(fmt.Errorf("storage '%s' in work queue no longer exists", key))
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	// Don't modify the informers copy
+	csr := original.DeepCopy()
+
+	reconcileErr := c.reconcile(ctx, csr)
+
+	if equality.Semantic.DeepEqual(original.Status, csr.Status) &&
+		equality.Semantic.DeepEqual(original.ObjectMeta, csr.ObjectMeta) {
+		// If we didn't change anything then don't call updateStatus.
+		// This is important because the copy we loaded from the informer's
+		// cache may be stale and we don't want to overwrite a prior update
+		// to status with this stale state.
+	} else if _, err := c.updateStatus(ctx, csr); err != nil {
+		// TODO: record the event (c.Recorder.Eventf(...
+		c.Logger.Warn("Failed to update PubSub Source status", zap.Error(err))
+		return err
+	}
+
+	if reconcileErr != nil {
+		// TODO: record the event (c.Recorder.Eventf(...
+		return reconcileErr
+	}
+
+	return nil
+}
+
+func (c *Reconciler) reconcile(ctx context.Context, ps *v1alpha1.PubSub) error {
+	ps.Status.ObservedGeneration = ps.Generation
+	// If topic has been already configured, stash it here
+	// since right below we remove them.
+	topic := ps.Status.TopicID
+
+	ps.Status.InitializeConditions()
+	// And restore it.
+
+	if topic == "" {
+		topic = ps.Spec.Topic
+	}
+
+	// See if the source has been deleted.
+	deletionTimestamp := ps.DeletionTimestamp
+
+	if deletionTimestamp != nil {
+		if err := c.PubSubBase.DeletePubSub(ctx, ps.Namespace, ps.Name); err != nil {
+			c.Logger.Infof("Unable to delete pubsub resources : %s", err)
+			return fmt.Errorf("failed to delete pubsub resources: %s", err)
+		}
+		c.removeFinalizer(ps)
+		return nil
+	}
+
+	// Ensure that there's finalizer there, since we're about to attempt to
+	// change external state with the topic, so we need to clean it up.
+	err := c.ensureFinalizer(ps)
+	if err != nil {
+		return err
+	}
+
+	t, psubs, err := c.PubSubBase.ReconcilePubSub(ctx, ps, topic, resourceGroup)
```

Comment:
The main role of the PubSub will be to propagate the status of the PullSubscription to this resource

Reply:
removed, thanks!